### PR TITLE
Fix app_offline detection for some OSes

### PR DESCRIPTION
**Pull request by BrennanConroy**
Reverts part of https://github.com/dotnet/aspnetcore/pull/49696. It looks like calling `WaitForMultipleObjects` isn't officially supported on a `HANDLE` from `CreateIoCompletionPort`. See https://stackoverflow.com/questions/39602908/waitforsingleobject-on-a-completion-port for some discussion on this.

On Windows Server 2012 R2 it looks like `ChangeNotificationThread` ends up hanging on `GetQueuedCompletionStatus` because `WaitForMultipleObjects` doesn't work properly with `m_hCompletionPort` and shutdown was triggered which causes `_lStopMonitorCalled` to be set so we don't end up calling `PostQueuedCompletionStatus` which would unblock `GetQueuedCompletionStatus`.

Customer issues https://github.com/dotnet/aspnetcore/issues/52307 and https://github.com/dotnet/aspnetcore/issues/52539, and possibly https://github.com/dotnet/aspnetcore/issues/52525

### CODE REVIEW

File: src/Servers/IIS/AspNetCoreModuleV2/RequestHandlerLib/filewatcher.cpp [hunk 1]
```diff
@@ -175,19 +169,8 @@ Win32 error
     LOG_INFO(L"Starting file watcher thread");
     DBG_ASSERT(pFileMonitor != nullptr);
 
-    HANDLE events[2] = { pFileMonitor->m_hCompletionPort, pFileMonitor->m_pShutdownEvent };
-
-    DWORD dwEvent = 0;
     while (true)
     {
-        // Wait for either a change notification or a shutdown event.
-        dwEvent = WaitForMultipleObjects(ARRAYSIZE(events), events, FALSE, INFINITE) - WAIT_OBJECT_0;
```
Comment (amcasey) [inline src/Servers/IIS/AspNetCoreModuleV2/RequestHandlerLib/filewatcher.cpp:184]: Would another mitigation be to reduce the wait time to something less than `INFINITE`?

**Comment by amcasey** [comment]
Cross-linking #45066

**Review by amcasey — APPROVED** [review]
😨

**Comment by adityamandaleeka** [comment]
Thanks @BrennanConroy for the investigation. Looks good at first glance but I'll give it a closer review today.

**Review by adityamandaleeka — APPROVED** [review]

**Comment by talanc-stramit** [comment]
Will this fix make it into the next version of .NET 8?

**Comment by BrennanConroy** [comment]
No, put it will make it into the February patch probably.

**Comment by bodenkers** [comment]
When will this be back ported for .net 8. We currently have aspnetcore 8.0.4 on our IIS servers and are still having this issue.

**Comment by adityamandaleeka** [comment]
8.0.4 should already have this fix. Are you sure it's this issue? Are you running on Windows Server 2012 R2?

**Comment by bodenkers** [comment]
We are experiencing it on  Microsoft Windows Server 2019 Standard. the bug
in mention stated that it could be happening on newer OS as well

**Comment by BrennanConroy** [comment]
Have you tried the workaround? https://github.com/dotnet/aspnetcore/issues/52307#issuecomment-1852886535

File a new issue, this is a closed PR.

**Comment by bodenkers** [comment]
Yes and it did not fix the problem

**Comment by bodenkers** [comment]
I have filed a new bug for this exact same behavior.
https://github.com/dotnet/aspnetcore/issues/55807

**Pull request merged**